File: Rnwood.Smtp4dev/ClientApp/ApiClient/MessageSummary.ts
 

export default class MessageSummary {

    constructor(id: string, from: string, to: string, receivedDate: Date, subject: string, attachmentCount: number, isUnread: boolean, ) {
         
        this.id = id; 
        this.from = from; 
        this.to = to; 
        this.receivedDate = receivedDate; 
        this.subject = subject; 
        this.attachmentCount = attachmentCount; 
        this.isUnread = isUnread;
    }

     
    id: string; 
    from: string; 
    to: string; 
    receivedDate: Date; 
    subject: string; 
    attachmentCount: number; 
    isUnread: boolean;
}

File: Rnwood.Smtp4dev.Shared/ApiModel/MessageSummary.cs
using System;

namespace Rnwood.Smtp4dev.ApiModel
{
    public class MessageSummary : ICacheByKey
    {

        public bool IsRelayed { get; set; }

        public Guid Id { get; set; }

        public string From { get; set; }
        public string To { get; set; }
        public DateTime ReceivedDate { get; set; }

        public string Subject { get; set; }

        public int AttachmentCount { get; set; }

        public bool IsUnread { get; set; }

        string ICacheByKey.CacheKey => Id.ToString() + IsUnread + IsRelayed;
    }
}
